lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -231,6 +231,8 @@ bool ReplaceOpenCLBuiltinPass::runOnFunction(Function &F) {
   case Builtins::kBarrier:
   case Builtins::kWorkGroupBarrier:
     return replaceBarrier(F);
+  case Builtins::kSubGroupBarrier:
```

Comment:
OpenCL defines a version of this function with an explicit memory scope operand. The memory scopes are already problematic (see #604). I'd suggesting leaving to the other issue to fix.

Reply:
work_group_barrier also has an explicit memory scope overload. Is that supported?

Reply:
I'll deal with that in #604 too. Certainly the scope operand isn't handled correctly.